core/vehicle/api.go

```diff
@@ -43,6 +43,13 @@ type API interface {
 	// SetPlanSoc sets the charge plan time and soc
 	SetPlanSoc(time.Time, int) error
 
+	// GetRepeatingPlans returns every repeating plan
+	GetRepeatingPlans(onlyActivePlans bool) []api.RepeatingPlanStruct
+	// SetRepeatingPlans stores every repeating plan
+	SetRepeatingPlans([]api.RepeatingPlanStruct) error
+	// GetRepeatingPlansAsNormalPlans returns every repeating plan with a timestamp-key to sort
+	GetRepeatingPlansWithTimestamps(onlyActivePlans bool) []api.PlanStruct
```

Comment:
Ich würde für diesen Daten-Zwischenstand keine neue öffentliche API am Fahrzeug machen. Das wird ja "nur" für die Berechnung des nächsten Zeitslots verwendet. Ich würd die Berechnung in der Methode verschieben, die rausfindet was die nächsten zu beachtenden Zeitpunkte sind. Also eher nen Aufruf, der in der Schleife bei `vehiclePlanSoc()` läuft.

Reply:
Du würdest also die Methode entfernen und den Code dieser Methode in die Methode `vehiclePlanSoc` verschieben?